File: api/src/main/java/com/api/app/controller/claim/ClaimController.java
package com.api.app.controller.claim;

import com.api.app.common.response.ApiResponse;
import com.api.app.common.security.SecurityUtils;
import com.api.app.dto.request.claim.CancelRequest;
import com.api.app.service.claim.ClaimService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.*;

import jakarta.validation.Valid;

/**
 * 클레임 컨트롤러
 *
 * @author Claude
 * @version 1.0
 * @since 2025-11-01
 */
@Slf4j
@RestController
@RequestMapping("/api/claim")
@RequiredArgsConstructor
@Tag(name = "클레임 관리", description = "주문 취소/반품/교환 API")
public class ClaimController {

    private final ClaimService claimService;
    private final SecurityUtils securityUtils;

    /**
     * 주문 취소
     *
     * @param request 주문 취소 요청
     * @return 성공 응답
     */
    @Operation(summary = "주문 취소", description = "주문을 취소합니다")
    @PostMapping("/cancel")
    public ApiResponse<Void> cancelOrder(@Valid @RequestBody CancelRequest request) {
        log.info("Cancel order request received. targetCount={}", request.getTargets().size());

        // 토큰에서 회원번호 추출
        String memberNo = securityUtils.getCurrentUserMemberNo();
        request.setMemberNo(memberNo);

        // 주문 취소 처리
        claimService.cancelOrder(request);

        log.info("Cancel order completed. memberNo={}", memberNo);

        return ApiResponse.success();
    }
}

File: api/src/main/kotlin/com/api/app/controller/claim/ClaimController.kt
package com.api.app.controller.claim

import com.api.app.common.response.ApiResponse
import com.api.app.common.security.SecurityUtils
import com.api.app.dto.request.claim.CancelRequest
import com.api.app.service.claim.ClaimService
import jakarta.validation.Valid
import org.slf4j.LoggerFactory
import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/claim")
@PreAuthorize("isAuthenticated()")
class ClaimController(
    private val claimService: ClaimService,
    private val securityUtils: SecurityUtils
) {

    private val log = LoggerFactory.getLogger(this::class.java)

    @PostMapping("/cancel")
    fun cancelOrder(@Valid @RequestBody request: CancelRequest): ApiResponse<Void> {
        val memberNo = securityUtils.getCurrentUserMemberNo()
        log.info("Cancel order request received. targetCount={}", request.targets.size)
        claimService.cancelOrder(request.copy(memberNo = memberNo))
        log.info("Cancel order completed. memberNo={}", memberNo)
        return ApiResponse.success()
    }
}
